### DISCUSSION TITLE
is Quick JS Useable with Consoles, Performance wise?

### DISCUSSION THREAD

**Discussion by JailbreakPapa** [quickjs-ng/quickjs#257] [Q&A]
Hey,

I'm currently exploring options for JS Engines for my HTML/CSS/JS Engine, and I'm currently looking for non-JIT based ones (since its console, mainly PS5), and i came across quickjs. I'm still investigating Hermes & V8, but both of those engines don't have a console port (obviously), so its taking me a bit to port them over. So, I'm wondering if this could be a nice fit for a decently fast JS Engine/Interp for consoles?

**Comment by saghul** [comment]
I think it's a good fit because it should be easy to port, given its small size.

**Comment by bnoordhuis** [comment]
QuickJS obviously isn't as fast as V8 but among the interpreters it's probably the fastest of the bunch.

And like Saúl said, it's pretty straightforward to port. Most likely only needs minor build tweaks. @TooTallNate ported it to the Nintendo Switch without much hassle, I believe.

By the way, speaking as someone who has worked with V8 a great deal and for a long time (15 years now?): V8 moves so fast, and is so big, it's practically impossible for a single individual to keep a port in good shape.

**Reply by TooTallNate** [reply]
Indeed! I am using QuickJS for a [JS runtime for Nintendo Switch](https://github.com/TooTallNate/nx.js/). I'm very satisfied with QuickJS and I'd imagine it would be a good fit on PS5 as well.
